src/kvstore.c

```diff
@@ -42,6 +42,7 @@ struct _kvstore {
     unsigned long long *dict_size_index;   /* Binary indexed tree (BIT) that describes cumulative key frequencies up until given dict-index. */
     size_t overhead_hashtable_lut;         /* The overhead of all dictionaries. */
     size_t overhead_hashtable_rehashing;   /* The overhead of dictionaries rehashing. */
+    void *metadata[];                      /* conditionally allocated based on "flags" */
```

Comment:
Hi @moticless I still think we don't need to create a new struct and add here and `kvstoreDictMetadata` , as we know, `key_count`, `overhead_hashtable_lut ` also are just statistics, extra metadata makes it more complex.

ping @guybe7 you bring the `kvstore` struct, WDYT?

Reply:
i didn't read the PR - can't we use the dict metadata of each dict inside the kvstore? or is that unnecessary and will take too much memory?

Reply:
The histogram is maintained at two levels. One per kvstore and one per slot. This gives future capability in clustering mode to quickly update kvstore/db level histogram when deleting a slot.

Whatever we decide in this thread. Please let's have implementation at a distinct PR. We need to cherry-pick this  commit ASAP into earlier version.